calculator/main.py

```diff
@@ -37,14 +50,18 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
-
+    if not isinstance(operation, dict) :
+        raise InvalidOperation('Given operation is invalid.')
+        
+    calc["operations"].update(operation)
```

Comment:
You could also check that there is only one key and value that what way somone isn't overloading a single operation like so:

```python
add_new_operation(calc, {'add': lambda1, lambda2, lambda3, etc...}
````